cns/NetworkContainerContract.go

```diff
@@ -338,26 +340,47 @@ type CreateNetworkContainerResponse struct {
 	Response Response
 }
 
-// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specifc network container.
+// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specific network container.
 type GetNetworkContainerStatusRequest struct {
 	NetworkContainerid string
 }
 
-// GetNetworkContainerStatusResponse specifies response of retriving a network container status.
+// GetNetworkContainerStatusResponse specifies response of retrieving a network container status.
 type GetNetworkContainerStatusResponse struct {
 	NetworkContainerid string
 	Version            string
 	AzureHostVersion   string
 	Response           Response
 }
 
-// GetNetworkContainerRequest specifies the details about the request to retrieve a specifc network container.
+// GetAllNetworkContainersRequest specifies the details about the request to retrieve all network containers on CNS.
+type GetAllNetworkContainersRequest struct {
+	NodeID string
+}
+
+// GetAllNetworkContainersResponse returns all network containers' ID on the node.
+type GetAllNetworkContainersResponse struct {
+	NetworkContainerIDs []string
```

Comment:
I was planning to check if there is any NC mis-matched or lost, by only comparing the NC IDs. Or even just simply compare the size/numbers of how many NCs that DNC & CNS each has.
However, if there are any occasions when the NC ID matches, but the other GS's properties do not match, then this will be terrible solution.
Hence it is better to return all details of NC. Approach updated.